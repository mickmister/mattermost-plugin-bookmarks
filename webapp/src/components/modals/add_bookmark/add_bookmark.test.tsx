// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow, mount, render} from 'enzyme';

import AddBookmarkModal from './add_bookmark';

describe('components/AddBookmark', () => {
    const baseActions = {
        getBookmark: jest.fn().mockResolvedValue({
            postID: 'ID1',
            title: 'my title',
            label_ids: ['labelid1'],
        }),
        getAllLabels: jest.fn().mockResolvedValue({
            ByID: {
                labelid1: {
                    name: 'id1name',
                },
                labelid2: {
                    name: 'id1name',
                },
            },

        }),
        close: jest.fn(),
        save: jest.fn(),
    };

    const baseProps = {
        ...baseActions,
        channelId: 'channelID',
        post: {
            id: 'asdfasdf',
            message: 'This is the post message',
        },
        visible: true,
    };

    it('snapshot modal not visible', () => {
        const props = {...baseProps, visible: false};
        const wrapper = shallow(<AddBookmarkModal {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('snapshot modal visible', () => {
        const props = {...baseProps};
        const wrapper = shallow(<AddBookmarkModal {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
