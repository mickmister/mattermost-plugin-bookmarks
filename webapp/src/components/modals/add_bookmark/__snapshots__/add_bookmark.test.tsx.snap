// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AddBookmark snapshot modal not visible 1`] = `
<Bootstrap(Modal)
  backdrop="static"
  bsSize="large"
  dialogClassName="modal--scroll"
  onExited={[Function]}
  onHide={[Function]}
  show={false}
  style={
    Object {
      "height": "100%",
    }
  }
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Edit Bookmark
    </ModalTitle>
  </ModalHeader>
</Bootstrap(Modal)>
`;

exports[`components/AddBookmark snapshot modal visible 1`] = `
<Bootstrap(Modal)
  backdrop="static"
  bsSize="large"
  dialogClassName="modal--scroll"
  onExited={[Function]}
  onHide={[Function]}
  show={true}
  style={
    Object {
      "height": "100%",
    }
  }
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Edit Bookmark
    </ModalTitle>
  </ModalHeader>
  <AddBookmarkForm
    channelId="channelID"
    close={[MockFunction]}
    getAllLabels={[MockFunction]}
    getBookmark={[MockFunction]}
    post={
      Object {
        "id": "asdfasdf",
        "message": "This is the post message",
      }
    }
    save={[MockFunction]}
    visible={true}
  />
</Bootstrap(Modal)>
`;
